ts/features/pn/store/reducers/__tests__/payments.test.ts

```diff
@@ -304,4 +319,390 @@ describe("PN Payments reducer's tests", () => {
+    const paymentStatus = paymentStatusForUISelector(
+      state,
+      "m1" as UIMessageId,
+      "p2"
+    );
+    expect(paymentStatus).toBe(remoteUndefined);
+  });
+  it("paymentStatusForUISelector should return remoteUndefined for a matching <message Id, payment Id> that is loading", () => {
+    const startingState = appReducer(undefined, {} as Action);
+    const updatePaymentForMessageAction = updatePaymentForMessage.request({
+      messageId: "m1" as UIMessageId,
+      paymentId: "p1"
+    });
+    const state = appReducer(startingState, updatePaymentForMessageAction);
+    const paymentStatusOnStore =
+      state.features.pn.payments["m1" as UIMessageId]?.p1;
+    expect(paymentStatusOnStore).toBe(remoteLoading);
+    const paymentStatus = paymentStatusForUISelector(
+      state,
+      "m1" as UIMessageId,
+      "p1"
+    );
+    expect(paymentStatus).toBe(remoteUndefined);
+  });
+  it("paymentStatusForUISelector should return remoteReady for a matching <message Id, payment Id> that is payable", () => {
+    const paymentData = {} as PaymentRequestsGetResponse;
+    const startingState = appReducer(undefined, {} as Action);
+    const updatePaymentForMessageAction = updatePaymentForMessage.success({
+      messageId: "m1" as UIMessageId,
+      paymentId: "p1",
+      paymentData
+    });
+    const state = appReducer(startingState, updatePaymentForMessageAction);
+    const paymentStatus = paymentStatusForUISelector(
+      state,
+      "m1" as UIMessageId,
+      "p1"
+    );
+    expect(paymentStatus).toStrictEqual(remoteReady(paymentData));
+  });
+  it("paymentStatusForUISelector should return remoteError for a matching <message Id, payment Id> that is not payable anymore", () => {
+    const details = Detail_v2Enum.PPT_PAGAMENTO_DUPLICATO;
+    const startingState = appReducer(undefined, {} as Action);
+    const updatePaymentForMessageAction = updatePaymentForMessage.failure({
+      messageId: "m1" as UIMessageId,
+      paymentId: "p1",
+      details
+    });
+    const state = appReducer(startingState, updatePaymentForMessageAction);
+    const paymentStatus = paymentStatusForUISelector(
+      state,
+      "m1" as UIMessageId,
+      "p1"
+    );
+    expect(paymentStatus).toStrictEqual(remoteError(details));
+  });
+  it("paymentsButtonStateSelector should return hidden for an unmatching message Id on store", () => {
+    const updatePaymentForMessageAction = updatePaymentForMessage.request({
+      messageId: "m1" as UIMessageId,
+      paymentId: "p1"
+    });
+    const startingState = appReducer(undefined, updatePaymentForMessageAction);
+    const buttonState = paymentsButtonStateSelector(
+      startingState,
+      "m2" as UIMessageId,
+      undefined,
+      5
+    );
+    expect(buttonState).toBe("hidden");
+  });
+  it("paymentsButtonStateSelector should return hidden when all visible payments are processed", () => {
+    const appState = appReducer(
+      appReducer(
```

Comment:
Should we use the `reproduceSequence` utility?

```Typescript
const sequenceOfActions: ReadonlyArray<Action> = [
  updatePaymentForMessage.failure({
    messageId: "m1" as UIMessageId,
    paymentId: "c1n1",
    details: Detail_v2Enum.PPT_PAGAMENTO_DUPLICATO
  }),
  updatePaymentForMessage.failure({
    messageId: "m1" as UIMessageId,
    paymentId: "c1n2",
    details: Detail_v2Enum.PPT_PAGAMENTO_DUPLICATO
  }),
  updatePaymentForMessage.failure({
    messageId: "m1" as UIMessageId,
    paymentId: "c1n3",
    details: Detail_v2Enum.PPT_PAGAMENTO_DUPLICATO
  }),
  updatePaymentForMessage.failure({
    messageId: "m1" as UIMessageId,
    paymentId: "c1n4",
    details: Detail_v2Enum.PPT_PAGAMENTO_DUPLICATO
  }),
  updatePaymentForMessage.failure({
    messageId: "m1" as UIMessageId,
    paymentId: "c1n5",
    details: Detail_v2Enum.PPT_PAGAMENTO_DUPLICATO
  })
];

const finalState = reproduceSequence(
  {} as GlobalState,,
  appReducer,
  sequenceOfActions
);
```

Reply:
Nice utility, applied